dev/MRTCore/packaging/MrtCore.PriGen.targets

```diff
@@ -1584,4 +1584,69 @@
       Platform="$(Platform)">
     </ValidateConfiguration>
   </Target>
+
+  <!-- Configure Fast Up To Date for PRI generation. Please see the comment in ProjectItemsSchema.xaml for more info. -->
+
+  <ItemGroup>
+    <PropertyPageSchema Include="$(MSBuildThisFileDirectory)ProjectItemsSchema.xaml" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <_MrtCoreUpToDateCheckInputDependsOn>
+      $(_MrtCoreUpToDateCheckInputDependsOn);
+      GetMrtPackagingOutputs;
+      _GetPriFilesFromPayload;
+    </_MrtCoreUpToDateCheckInputDependsOn>
+  </PropertyGroup>
+
+  <!-- Inform the fast up-to-date check of the inputs for PRI generation. -->
+  <Target Name="_MrtCoreUpToDateCheckInput" DependsOnTargets="$(_MrtCoreUpToDateCheckInputDependsOn)" BeforeTargets="CollectUpToDateCheckInputDesignTime">
+        <!-- 
+        Copied from the target _GenerateProjectPriConfigurationFiles. @(_LayoutFile),  @(_EmbedFile), and @(_PRIResourceFiltered) are ultimately inputs to
+        the MRT Core indexer, but they're populated by _GenerateProjectPriConfigurationFiles which does work (runs a tool to generate the indexer
+        configuration files) that we don't need during a design-time build, so we hoist them out to avoid calling the target.
+        -->
+    <ItemGroup>
+        <!--
+        First, build out the complete list of files we want to consider for the layout. Then exclude anything that matches any pattern or filename listed in
+        _AppxLayoutAssetPackageFiles. We could do this as a 'Remove' operation, but by building an oracle we don't modify, we simplify future manipulations of
+        this data set.
+        -->
+      <_LayoutFileSource Include="@(PackagingOutputs)" Condition="'%(OutputGroup)' == 'ContentFilesProjectOutputGroup' and '%(ProjectName)' == '$(ProjectName)'" />
+      <_LayoutFileSource Include="@(PackagingOutputs)" Condition="'%(OutputGroup)' == 'CustomOutputGroupForPackaging' and '%(ProjectName)' == '$(ProjectName)'" />
+      <_LayoutFile Include="@(_LayoutFileSource)" Exclude="@(_AppxLayoutAssetPackageFiles)" />
+
+      <_EmbedFile Include="@(PackagingOutputs)" Condition="'%(OutputGroup)' == 'EmbedOutputGroupForPackaging' and '%(ProjectName)' == '$(ProjectName)'"/>
+
+      <!-- If we have the .xbf, we don't need the .xaml file. -->
+      <_LayoutFileXbfXaml Include="$([System.IO.Path]::ChangeExtension('%(_LayoutFile.Identity)','.xaml'))" Condition="'%(Extension)' == '.xbf'" />
+      <_LayoutFile Remove="@(_LayoutFileXbfXaml)" />
+
+      <!-- Filter out PRIResource files which are marked by C++ project system as ExcludedFromBuild. -->
+      <_PRIResourceFiltered Include="@(PRIResource)" Condition="'%(PRIResource.ExcludedFromBuild)' != 'true'" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <_UpToDateCheckInputTemp Include="@(SourceAppxManifest)" Set="PRIGeneration" />
+      <_UpToDateCheckInputTemp Include="@(_LayoutFile)" Condition="'%(Extension)'!='.xbf'" Set="PRIGeneration" />
```

Comment:
>'%(Extension)'!='.xbf' [](http://example.com/codeflow?start=67&length=22)

is xbf always embedded?

Reply:
I believe so. @evelynwu-msft to confirm.

Reply:
There are cases where they are not, but by definition those are consequently not inputs to resources.pri.

Reply:
if xbf is not embedded, wouldn't changing of xbf require a build to update the file?

Reply:
(to be honest, I don't recall why I added this condition)

Reply:
.xbf files will only change if source .xaml or code-behind are modified, and this will already trigger a build independent of what MRT Core does/doesn't do.

Reply:
then do we need to do the extra remove from line 1621-1623?

Reply:
If we have a loose XBF, then we're not going to embed it (I believe this is our intended design). We exclude XBFs in the input set for that reason, so excluding the corresponding XAMLs too makes sense to me. @evelynwu-msft is all that correct?

Reply:
I could imagine a weird edge cases where a developer embeds a `.xbf` that was externally generated by including it in the package as part of `@(Content)` In that case, you'd want a change to it to be recognized as making `resources.pri` out of date.

Reply:
@huichen123 Spoke with Evelyn offline. I'll remove the XBF filtering in `<_UpToDateCheckInputTemp Include="@(_LayoutFile)" Condition="'%(Extension)'!='.xbf'" Set="PRIGeneration" />` and leave the _conditional_ XAML removal as it is. Any objections?

Reply:
sounds good to me.